api_tests/files/views/test_file_detail.py

```diff
@@ -602,6 +628,19 @@ def test_current_version_is_equal_to_length_of_history(
             res = app.get(file_url, auth=user.auth)
             assert res.json['data']['attributes']['current_version'] == version
 
+    def test_version_nonsequential(self, app, user, nonsequential_version_url, nonsequential_file):
+        """
+        When a user makes concurrent requests to make a new version they can produce two or more versions with the same
+        identifier. This test insures all versions are returned when queried. This behavior may change so version
+        identifiers are unique, in that case it's okay to delete this test.
+        """
+        res = app.get(nonsequential_version_url, auth=user.auth)
+        data = res.json['data']
+        assert len(data) == 2
+
+        assert data[0]['id'] == '1'
+        assert data[1]['id'] == '1'
```

Comment:
Very nice. Could you also test that `get_version` and `find_child_by_name` return what you expect it to?

Reply:
Yes and I've added a new test for `find_child_by_name` because there wasn't one initially.